pkg/cmd/pulumi/new.go

```diff
@@ -813,6 +846,102 @@ func saveConfig(stack backend.Stack, c config.Map) error {
 	return saveProjectStack(stack, ps)
 }
 
+func runtimeOptions(ctx *plugin.Context, info *workspace.ProjectRuntimeInfo,
+	main string, opts display.Options, yes, interactive bool, prompt promptForValueFunc,
+) (map[string]interface{}, error) {
+	programInfo := plugin.NewProgramInfo(ctx.Root, ctx.Pwd, main, info.Options())
+	lang, err := ctx.Host.LanguageRuntime(info.Name(), programInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load language plugin %s: %w", info.Name(), err)
+	}
+
+	options := make(map[string]interface{}, len(info.Options()))
+	for k, v := range info.Options() {
+		options[k] = v
+	}
+
+	// Keep querying for prompts until there are no more.
+	for {
+		// Update the program info with the latest runtime options.
+		programInfo := plugin.NewProgramInfo(ctx.Root, ctx.Pwd, main, options)
+		prompts, err := lang.RuntimeOptionsPrompts(programInfo)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get runtime options prompts: %w", err)
+		}
+
+		if len(prompts) == 0 {
+			break
+		}
+
+		surveycore.DisableColor = true
+		for _, optionPrompt := range prompts {
+			if len(optionPrompt.Choices) == 1 {
+				// We got exactly one choice, so use that as default value without prompting the user.
+				choice := optionPrompt.Choices[0]
+				options[optionPrompt.Key] = choice.Value()
+			} else if len(optionPrompt.Choices) > 0 {
+				// Pick one among the choices
+				choices := make([]string, 0, len(optionPrompt.Choices))
+				// Map choice display string to the actual value
+				choiceMap := make(map[string]interface{}, len(optionPrompt.Choices))
+				for _, choice := range optionPrompt.Choices {
+					choices = append(choices, choice.String())
+					choiceMap[choice.String()] = choice.Value()
+				}
+
+				if !interactive {
+					// In non-interactive mode pick the first option
+					options[optionPrompt.Key] = choiceMap[choices[0]]
+				} else {
+					var response string
+					message := opts.Color.Colorize(colors.SpecPrompt + "\r" + optionPrompt.Description + colors.Reset)
+					if err := survey.AskOne(&survey.Select{
+						Message: message,
+						Options: choices,
+					}, &response, surveyIcons(opts.Color), nil); err != nil {
+						return nil, err
+					}
+					options[optionPrompt.Key] = choiceMap[response]
+				}
+			} else {
+				// Free form input
+				if !interactive {
+					// Skip free form input in non-interactive mode
```

Comment:
Should this be an error instead?

Reply:
Maybe, I wonder if the multiple choice case above should be an error as well? 
That or make it clear to langruntimes that the contract is that choice order is important, and freeform must be optional.

Reply:
We want to default to something here, so that we can maintain backwards compatibility if someone os running a scripted version of `pulumi new some-python-template --yes --non-interactive --name ...`.

Maybe a more explicit solution would be add a `Default: RuntimeOptionValue` for each prompt. If we are in non-interactive mode, the default is used (for choices and free form input), and error out if there is no default.

Reply:
Yeh that feels ok

Reply:
Updated to provide an optional default https://github.com/pulumi/pulumi/compare/a66cb73d51c77d399c94aebb37aaff28bf6c8547..720bcdc9c148705a51893ad7e484b5d2ad446f55